Neos.ContentRepository.Core/Classes/EventStore/EventPersister.php

```diff
@@ -23,18 +28,18 @@
 {
     public function __construct(
         private EventStoreInterface $eventStore,
-        private ProjectionCatchUpTriggerInterface $projectionCatchUpTrigger,
         private EventNormalizer $eventNormalizer,
-        private Projections $projections,
+        private ProjectionsAndCatchUpHooks $projectionsAndCatchUpHooks
     ) {
     }
 
     /**
      * @param EventsToPublish $eventsToPublish
+     * @param ContentRepository $contentRepository only required to build the possible catchup hooks
      * @return CommandResult
      * @throws ConcurrencyException in case the expectedVersion does not match
      */
-    public function publishEvents(EventsToPublish $eventsToPublish): CommandResult
+    public function publishEvents(EventsToPublish $eventsToPublish, ContentRepository $contentRepository): CommandResult
```

Comment:
todo rename and do something usefull with `CommandResult`

Reply:
result will be removed via https://github.com/neos/neos-development-collection/pull/5061